src/main/java/ru/ayaz/User.java

```diff
@@ -0,0 +1,54 @@
+package ru.ayaz;
+
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.security.Principal;
+
+/**
+ * Created by ayaz on 11/5/16.
+ */
+public class User {
+    private String nickName;
+    private long id;
+    private PrintWriter writer;
```

Comment:
TreeMap is a map with "id" key and user **reference**. Each user object has "writer" field to be written message to.